docs/CODE_GUIDELINES.md

```diff
@@ -632,6 +633,31 @@ class Foo
 
 A class with any virtual functions should have a destructor that is either public and virtual or else protected and nonvirtual (cf. [ISO C++ guidelines](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-dtor-virtual)).
 
+### 11.6. Constructor Initialzation Lists
+
+For short lines everything on one line is OK, excluding the braces which must be on the following lines.
+
+```cpp
+MyClass::CMyClass(bool bBoolArg, int iIntegerArg) : m_bArg(bBoolArg), m_iArg(iIntegerArg)
+{
+}
+```
+
+For longer lines, break before colon and after comma.
+
+```cpp
+MyClass::CMyClass(bool bBoolArg,
+                  int iIntegerArg,
+                  const std::string& strSomeText,
+                  const std::shared_ptr<CMyOtherClass>& myOtherClass)
+  : m_bBoolArg(bBoolArg),
+    m_iIntegerArg(iIntegerArg),
+    m_strSomeText(strSomeText),
+    m_myOtherClass(myOtherClass)
```

Comment:
The reason I like the following style (`BreakConstructorInitializers: BeforeComma`) is that in most cases it doesn't touch the previous line to add a comma if another initializer is added. If the majority prefers to keep the comma on the previous line that's okay for me.

```
MyClass::CMyClass(bool bBoolArg,
                  int iIntegerArg,
                  const std::string& strSomeText,
                  const std::shared_ptr<CMyOtherClass>& myOtherClass)
  : m_bBoolArg(bBoolArg)
  , m_iIntegerArg(iIntegerArg)
  , m_strSomeText(strSomeText)
  , m_myOtherClass(myOtherClass)
```

Reply:
Is "doesn't touch the previous line to add a comma if another initializer is added" really worth  this very special formatting style? My personal opinion: It's not.

Reply:
How is it "very special"?

Reply:
Only my personal feeling, which got inspired from other C++ projects I know.

Reply:
I think Rechi has a good point about keeping line history clean. However, I'm also with ksooo on this specific case, where I feel the leading comma is ugly enough to warrant an exception to the clean line mantra.

Reply:
I strongly prefer the leading comma because it provides much cleaner diffs.